### ISSUE TITLE
Laravel 9 and Flysystem 2/3 support

### ISSUE THREAD

**Issue by mmachatschek** [spatie/laravel-google-cloud-storage#2]
In the composer.json Laravel 9 was marked for installable dependencies by @riasvdv but unfortunatly this is not possible because the used libraries `spatie/flysystem-google-cloud-storage` and `league/flysystem-cached-adapter` only support flysystem version 1.

The optimal way to solve this issue would be to remove the dependencies  `spatie/flysystem-google-cloud-storage` and `league/flysystem-cached-adapter` and implement the GCS flysystem driver directly in this package the same way Laravel extended the AwsS3V3Adapter.

I already did some work on this but I'm not sure if this is of interest for you. PR#1

**Comment by riasvdv** [comment]
> In the composer.json Laravel 9 was marked for installable dependencies by @riasvdv but unfortunatly this is not possible because the used libraries spatie/flysystem-google-cloud-storage and league/flysystem-cached-adapter only support flysystem version 1.

Yes, that's why we haven't tagged a release yet as we noticed it would require a bit more work

> I already did some work on this but I'm not sure if this is of interest for you. PR #1

Thanks a lot for this!  We'll look into it soon and certainly discuss it further there
